### PR TITLE
Add COMPOPTS and EXECENV for HDF5 tests

### PR DESCRIPTION
**Pull request by daviditen**
Add a COMPOPTS to set include directories and lib directories for HDF5 and add -lhdf5 to the compile command.

Add an EXECENV to set the LD_LIBRARY_PATH for HDF5.

Update the directory-wide .skipif file to make it work with both XC systems and chapdl.

### CODE REVIEW

File: test/library/packages/HDF5/EXECENV [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+INSTALL_DIR=`h5cc -showconfig | sed -n -e 's/ *Installation point: \(.*\)/\1/p'`
+
+echo "LD_LIBRARY_PATH=$INSTALL_DIR/lib"
```
Comment (ronawho) [inline test/library/packages/HDF5/EXECENV:5]: Should this prefix `LD_LIBRARY_PATH` instead of overriding?

**Comment by daviditen** [comment]
Thanks! I updated the EXECENV file to prepend to LD_LIBRARY_PATH.

File: test/library/packages/HDF5/EXECENV [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+INSTALL_DIR=`h5cc -showconfig | sed -n -e 's/ *Installation point: \(.*\)/\1/p'`
+
+echo "LD_LIBRARY_PATH=$INSTALL_DIR/lib:LD_LIBRARY_PATH"
```
Comment (ronawho) [inline test/library/packages/HDF5/EXECENV:5]: `:LD_LIBRARY_PATH` -> `:$LD_LIBRARY_PATH`

**Review by ronawho — APPROVED** [review]

File: test/library/packages/HDF5/EXECENV (hunk 2 above)
Comment (daviditen) [reply test/library/packages/HDF5/EXECENV]: Oops. Thanks.

**Pull request merged**